### DISCUSSION TITLE
Deploying and hosting my bot

### DISCUSSION THREAD

**Discussion by RealA10N** [Rapptz/discord.py#6394] [General] [answered]
Hi, you'll! 😄 
I started developing my first discord bot, and I'm somewhat new to hosting my code.
I would like to know and learn from others experience: **What is the right way to host and manage a discord bot?**
I use Microsoft's Azure cloud services for all of my other projects, so it will be great if there is a simple way I can deploy my bot over there.
Thank you in advance!

**Comment by bandwiches** [comment]
If you want mobility, your best bet is probably a container.  AWS, Azure, and GCP all support containers in some way, you may have to tweak it a little but it works.  You can also look into using a Raspberry Pi if you'd rather keep it in-house.

**Comment by Rapptz** [comment, marked as answer]
A Discord bot is just a script that you run in the Python interpreter so any way to do that in your provider of choice will work just fine. You can do containers if you want to isolate it or you can just run it bare. The choice is mostly yours. There are no real constraints in how the library expects to be run, unlike e.g. a web server.

**Reply by RealA10N** [reply]
Thank you for your answer! I just wanted to learn from your experience. I guess I will go with a container then 😄
